src/PowerShell/Microsoft.WinGet.Client.Cmdlets/Cmdlets/AddSourceCmdlet.cs

```diff
@@ -42,13 +43,42 @@ public sealed class AddSourceCmdlet : PSCmdlet
             ValueFromPipelineByPropertyName = true)]
         public string Type { get; set; }
 
+        /// <summary>
+        /// Gets or sets the trust level of the source to add.
+        /// </summary>
+        [Parameter(ValueFromPipelineByPropertyName = true)]
+        public PSSourceTrustLevel TrustLevel { get; set; } = PSSourceTrustLevel.Default;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the source to add is explicit.
+        /// </summary>
+        ///
+        [Parameter(ValueFromPipelineByPropertyName = true)]
+        public SwitchParameter Explicit { get; set; }
+
         /// <summary>
         /// Adds source.
         /// </summary>
         protected override void ProcessRecord()
         {
             var command = new CliCommand(this);
-            command.AddSource(this.Name, this.Argument, this.Type);
+            command.AddSource(this.Name, this.Argument, this.Type, this.ConvertPSSourceTrustLevelToString(this.TrustLevel), this.Explicit.ToBool());
+        }
+
+        private string ConvertPSSourceTrustLevelToString(PSSourceTrustLevel trustLevel)
+        {
+            string trustLevelString = string.Empty;
+            switch (trustLevel)
+            {
+                case PSSourceTrustLevel.None:
+                    trustLevelString = "None";

```

Comment:
You could do switch return with PSSourceTrusLevel.Default => string.Empty and _ being trustLevel.ToString()

Reply:
Changed it to your fancy switch pattern syntax.